### PR TITLE
F# project file loading

### PR DESCRIPTION
**Pull request by surban**
F# project handling (`FSharpCompilation`, `FSharpDocument`, `FSharpProject`) has been added into the new project `Microsoft.DocAsCode.Metadata.ManagedReference.FSharp`. 

It uses [Dotnet.ProjInfo](https://www.nuget.org/packages/Dotnet.ProjInfo) to parse F# project files and the F# compiler service to access the internal compiler representation of the code.

Code for extract Metadata will be provided in a separate PR.

### CODE REVIEW

**Comment by superyyrrzz** [comment]
```
[21:57:07][Step 3/3] "C:\BuildAgent\work\130feb8a19fc2a14\docfx\All.sln" (Build target) (1) ->
[21:57:07][Step 3/3] "C:\BuildAgent\work\130feb8a19fc2a14\docfx\src\Microsoft.DocAsCode.Metadata.ManagedReference.FSharp\Microsoft.DocAsCode.Metadata.ManagedReference.FSharp.fsproj" (default target) (23) ->
[21:57:07][Step 3/3] (CoreCompile target) -> 
[21:57:07][Step 3/3]   FSC : error FS0226: The file extension of 'C:\BuildAgent\work\130feb8a19fc2a14\docfx\TEMP\version.cs' is not recognized. Source files must have extension .fs, .fsi, .fsx, .fsscript, .ml or .mli. [C:\BuildAgent\work\130feb8a19fc2a14\docfx\src\Microsoft.DocAsCode.Metadata.ManagedReference.FSharp\Microsoft.DocAsCode.Metadata.ManagedReference.FSharp.fsproj]
[21:57:07][Step 3/3] 
```

The PR validation fails because `Microsoft.DocAsCode.Metadata.ManagedReference.FSharp.fsproj` references `Shared\base.props`, which references `version.cs`. It failed because `.cs` is not compatible with `.fsproj`.

`version.cs` is generated by `build.ps1`. To unblock this PR, we will first modify `base.props` to be able comsume `version.fs`, and `build.ps` to generate `version.fs`.

**Comment by surban** [comment]
I have made the necessary changes.

File: src/Microsoft.DocAsCode.Metadata.ManagedReference.FSharp/FSharpProject.fs [hunk 1]
```diff
@@ -0,0 +1,86 @@
+﻿namespace Microsoft.DocAsCode.Metadata.ManagedReference.FSharp
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference.FSharp/FSharpProject.fs]: please add copyright header

File: src/Microsoft.DocAsCode.Metadata.ManagedReference.FSharp/FSharpProject.fs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+open System.IO
+open System.Collections.Generic
+
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+open Microsoft.DocAsCode.Metadata.ManagedReference
+
+
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference.FSharp/FSharpProject.fs]: remove extra blank lines

**Comment by surban** [comment]
Rebased to dev branch as requested.

**Pull request merged**